src/nevermined/api/RegistryBaseApi.ts

```diff
@@ -0,0 +1,190 @@
+import { NVMBaseApi } from "./NVMBaseApi"
+import { ServiceType } from "../../ddo/Service"
+import { MetaData, Account, DDO } from "../.."
+import { NvmConfigVersions } from "../../ddo/NvmConfig"
+import { TxParameters } from "../../keeper/contracts/ContractBase"
+import { SubscribablePromise, zeroX, generateId } from "../../utils"
+import { PublishMetadata } from "./AssetsApi"
+import { OrderProgressStep, UpdateProgressStep } from "../ProgessSteps"
+import { AssetError } from "../../errors/AssetError"
+
+/**
+ * It described the policy to be used when resolving an asset. It has the following options:
+ * * ImmutableFirst - It checks if there is a reference to an immutable data-store (IPFS, Filecoin, etc) on-chain. If that's the case uses the URL to resolve the Metadata. If not try to resolve the metadata using the URL of the Metadata/Marketplace API
+ * * MetadataAPIFirst - Try to resolve the metadata from the Marketplace/Metadata API, if it can't tries to resolve using the immutable url
+ * * OnlyImmutable - Try to resolve the metadata only from the immutable data store URL
+ * * OnlyMetadataAPI - Try to resolve the metadata only from the Marketplace/Metadata API
+ */
+export enum DIDResolvePolicy {
+    ImmutableFirst,
+    MetadataAPIFirst,
+    OnlyImmutable,
+    OnlyMetadataAPI
+}
+
+/**
+ * Abstract class proving common functionality related with Assets registration.
+ */
+export abstract class RegistryBaseApi extends NVMBaseApi {
```

Comment:
Is there a need for both `RegistryBaseApi` and `NVMBaseApi`?

Reply:
removing `NVMBaseApi`